src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -415,9 +433,12 @@ private static InetAddress getNetworkInterfaceAddress(String intf, String config
         }
     }
 
-    private static void setConfig(Config config)
+    private static void setConfig(Supplier<Config> config)
     {
-        conf = config;
+        conf = config.get();
+        confValueAccessors = ImmutableMap.copyOf(
+            withReplacementsLoader(
+                validatedPropertyLoader(p -> p.getAnnotation(Mutable.class) == null)).flatten(Config.class));
```

Comment:
IMO its cleaner to push this property into the loader itself, the loader 100% makes assumptions about this interface, but what is selected is external just adds extra confusion.

Reply:
I have simplified the @Mutable usage, so we don't need it now.
Removed.